src/client/components/Signup.tsx

```diff
@@ -12,10 +12,28 @@ interface SignUpProps{
 }
 
 export const SignUp: React.FC<SignUpProps> = () => {
+
     const [name, setName] = useState("");
     const [userName, setUserName] = useState("");
     const [password, setPassword] = useState("");
+    const [confirmedPassword, setConfirmedPassword] = useState("");
     const [email, setEmail] = useState("");
+    const [confirmedEmail, setConfirmedEmail] = useState("");
+
+    const areSame = (confirmed: string, regular: string) => {
+        return confirmed == regular || confirmed === "";
+    }
+
+    const properEmail = () => {
+        const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        console.log(re.test(String(email).toLowerCase()));
+        return re.test(String(email).toLowerCase()) || email === "";
+    }
+
+    const properPassword = () => {
+        return !((password.length < 9 || password.length > 255) && password.length > 0); 
```

Comment:
Revise this to make the conditional more clear. I know this works, but it is very unclear. Like I said before, try making this more affirmative such as creating a condition that should evaluate to true.